### PR TITLE
Add information about debugging failed tests.

### LINKED ISSUE #45: Provide information to debug failed deployments
A lot of times our tests will fail because we timeout waiting for a K8s deployment. 
An example is when a deployment fails to start because the image couldn't be pulled.

To debug the test failure we'd like to capture the events for pod because this provides helpful debug information.

For example here is the output of kubectl describe for a pod that failed to start because the name of the binary to invoke was wrong.

```
kubectl -n kubeflow-presubmit-kubeflow-e2e-299-439a983-360-cee1 describe pods tf-job-operator-7899645f6c-cgxd2
Name:           tf-job-operator-7899645f6c-cgxd2
Namespace:      kubeflow-presubmit-kubeflow-e2e-299-439a983-360-cee1
Node:           gke-kubeflow-testing-default-pool-3cd81eea-2dp8/10.142.0.2
Start Time:     Mon, 26 Feb 2018 18:02:23 -0800
Labels:         name=tf-job-operator
                pod-template-hash=3455201927
Annotations:    kubernetes.io/created-by={"kind":"SerializedReference","apiVersion":"v1","reference":{"kind":"ReplicaSet","namespace":"kubeflow-presubmit-kubeflow-e2e-299-439a983-360-cee1","name":"tf-job-operator-789...
Status:         Running
IP:             10.36.4.231
Created By:     ReplicaSet/tf-job-operator-7899645f6c
Controlled By:  ReplicaSet/tf-job-operator-7899645f6c
Containers:
  tf-job-operator:
    Container ID:  docker://e1d6dfd6955d2c4c1d89030626e8f143293f14bd55081421ce0d3da26ab64845
    Image:         gcr.io/kubeflow-images-staging/tf_operator:v20180226-403
    Image ID:      docker-pullable://gcr.io/kubeflow-images-staging/tf_operator@sha256:6c80d00498103a87d2acbdad34ce56886bacf29fb819e4424fa909dfc4c5b2e6
    Port:          <none>
    Command:
      /opt/mlkube/tf_operator
      --controller-config-file=/etc/config/controller_config_file.yaml
      --alsologtostderr
      -v=1
    State:       Waiting
      Reason:    CrashLoopBackOff
    Last State:  Terminated
      Reason:    ContainerCannotRun
      Message:   oci runtime error: container_linux.go:247: starting container process caused "exec: \"/opt/mlkube/tf_operator\": stat /opt/mlkube/tf_operator: no such file or directory"

      Exit Code:    127
      Started:      Mon, 26 Feb 2018 18:18:16 -0800
      Finished:     Mon, 26 Feb 2018 18:18:16 -0800
    Ready:          False
    Restart Count:  8
    Environment:
      MY_POD_NAMESPACE:  kubeflow-presubmit-kubeflow-e2e-299-439a983-360-cee1 (v1:metadata.namespace)
      MY_POD_NAME:       tf-job-operator-7899645f6c-cgxd2 (v1:metadata.name)
    Mounts:
      /etc/config from config-volume (rw)
      /var/run/secrets/kubernetes.io/serviceaccount from tf-job-operator-token-kk7pm (ro)
Conditions:
  Type           Status
  Initialized    True 
  Ready          False 
  PodScheduled   True 
Volumes:
  config-volume:
    Type:      ConfigMap (a volume populated by a ConfigMap)
    Name:      tf-job-operator-config
    Optional:  false
  tf-job-operator-token-kk7pm:
    Type:        Secret (a volume populated by a Secret)
    SecretName:  tf-job-operator-token-kk7pm
    Optional:    false
QoS Class:       BestEffort
Node-Selectors:  <none>
Tolerations:     node.alpha.kubernetes.io/notReady:NoExecute for 300s
                 node.alpha.kubernetes.io/unreachable:NoExecute for 300s
Events:
  Type     Reason                 Age                From                                                      Message
  ----     ------                 ----               ----                                                      -------
  Normal   Scheduled              19m                default-scheduler                                         Successfully assigned tf-job-operator-7899645f6c-cgxd2 to gke-kubeflow-testing-default-pool-3cd81eea-2dp8
  Normal   SuccessfulMountVolume  19m                kubelet, gke-kubeflow-testing-default-pool-3cd81eea-2dp8  MountVolume.SetUp succeeded for volume "config-volume"
  Normal   SuccessfulMountVolume  19m                kubelet, gke-kubeflow-testing-default-pool-3cd81eea-2dp8  MountVolume.SetUp succeeded for volume "tf-job-operator-token-kk7pm"
  Normal   Pulled                 18m (x4 over 19m)  kubelet, gke-kubeflow-testing-default-pool-3cd81eea-2dp8  Container image "gcr.io/kubeflow-images-staging/tf_operator:v20180226-403" already present on machine
  Normal   Created                18m (x4 over 19m)  kubelet, gke-kubeflow-testing-default-pool-3cd81eea-2dp8  Created container
  Warning  Failed                 18m (x4 over 19m)  kubelet, gke-kubeflow-testing-default-pool-3cd81eea-2dp8  Error: failed to start container "tf-job-operator": Error response from daemon: oci runtime error: container_linux.go:247: starting container process caused "exec: \"/opt/mlkube/tf_operator\": stat /opt/mlkube/tf_operator: no such file or directory"
  Warning  BackOff                9m (x44 over 19m)  kubelet, gke-kubeflow-testing-default-pool-3cd81eea-2dp8  Back-off restarting failed container
  Warning  FailedSync             4m (x74 over 19m)  kubelet, gke-kubeflow-testing-default-pool-3cd81eea-2dp8  Error syncing pod

```

### PR DESCRIPTION
**Pull request by jlewi**
* Add information about debugging failed deployments by getting K8s events.
* Add information about using Stackdriver logging to get pod logs for prow
  job.
* Fix #45

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -116,6 +113,47 @@ The ${POD_ID} is of the form
 ${WORKFLOW_ID}-${RANDOM_ID}
 ```
 
+## Debuging Failed Tests
```
Comment (lluunn) [inline README.md]: debugging

File: README.md [hunk 2]
```diff
@@ -89,17 +89,14 @@ The argo UI is publicly accessible at http://testing-argo.kubeflow.org/timeline.
...
-  * [argo/issues#710](https://github.com/argoproj/argo/issues/710) exit handlers aren't shown
-
-So if your exit handler fails you may need to look at pod logs or Stackdriver logs directly.
-
 ### Stackdriver logs
 
 Since we run our E2E tests on GKE, all logs are persisted in Stackdriver logging.
 
-Access to Stackdriver logs is restricted. We are working on giving sufficient access to members of the community ([kubeflow/testing#5](https://github.com/kubeflow/testing/issues/5)).
+Viewer access to Stackdriver logs is available by joining one of the following groups
```
Comment (ankushagarwal) [inline README.md]: Add link to stackdriver

https://console.cloud.google.com/logs/viewer?project=kubeflow-ci&resource=k8s_cluster%2Flocation%2Fus-east1-d%2Fcluster_name%2Fkubeflow-testing

**Pull request merged**